pxtblocks/code-validation/rubric.ts

```diff
@@ -0,0 +1,30 @@
+namespace pxt.blocks {
+    export class Rubric {
```

Comment:
I like the architecture that you've set up for the rubric and the criteria. It's easy to understand and it will be pretty easy to expand on, too. Nice!